src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/AddImport/AddImportPlacementOptionsProviders.cs

```diff
@@ -4,21 +4,32 @@
 
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
 
 namespace Microsoft.CodeAnalysis.AddImport;
 
-internal static partial class AddImportPlacementOptionsProviders
+internal static class AddImportPlacementOptionsProviders
 {
-    internal static async ValueTask<AddImportPlacementOptions> GetAddImportPlacementOptionsAsync(this Document document, IAddImportsService addImportsService, CancellationToken cancellationToken)
-    {
+    // Normally we don't allow generation into a hidden region in the file.  However, if we have a
+    // modern span mapper at our disposal, we do allow it as that host span mapper can handle mapping
+    // our edit to their domain appropriate.
+    public static bool AllowImportsInHiddenRegions(this Document document)
 #if CODE_STYLE
-        var syntaxTree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-        var configOptions = document.Project.AnalyzerOptions.AnalyzerConfigOptionsProvider.GetOptions(syntaxTree).GetOptionsReader();
-        return addImportsService.GetAddImportOptions(configOptions, allowInHiddenRegions: false);
+        => AddImportPlacementOptions.Default.AllowInHiddenRegions;
 #else
-        return await document.GetAddImportPlacementOptionsAsync(cancellationToken).ConfigureAwait(false);
+        => document.Services.GetService<Host.ISpanMappingService>()?.SupportsMappingImportDirectives == true;
 #endif
+
+#if !CODE_STYLE
```

Comment:
still needed?  will it be possible to remove at some point?

Reply:
Yes, still needed. One problem is that specific language services (such as IAddImportsService) are internal.

Reply:
It would be possible if we made more APIs public.